IdentityCore/src/cache/token/MSIDCredentialCacheItem.m

```diff
@@ -323,6 +334,10 @@ - (BOOL)matchesWithRealm:(nullable NSString *)realm
             return NO;
         }
     }
+    if (credentialType && self.credentialType != credentialType)
```

Comment:
can you make sure this won't cause any issues? e.g. querying refresh token should also return legacyrefreshtoken, where this condition would fail. Also we already have a type keychain query attribute. Wouldn't that be enough?

Reply:
Removed this logic. We already pass the credential Type as key.type which will be different from AccessToken and MSIDAccessTokenWithAuthScheme for querying access token so I wont need this additional matching. NSNumber *type = key.type;